include/caffe/layer.hpp

```diff
@@ -316,6 +317,21 @@ class Layer {
     param_propagate_down_[param_id] = value;
   }
 
+  bool IsForwardPassed() const {
```

Comment:
(Perhaps the intention here was to keep track not simply of whether an individual layer's forward has completed, but whether _all_ layers' forwards' have completed. But note that that's not what this code does anyway.)